File: eTierMeeting_v2/API/Helpers/Params/SearchHistoryParams.cs

namespace Machine_API.Helpers.Params
{
    public class SearchHistoryParams
    {
        public string MachineId { get; set; }
        public int? PdcId { get; set; }
        public int? BuildingCode { get; set; }
        public int? CellCode { get; set; }
        public string PositionCode { get; set; }
        public string TimeStart { get; set; }
        public string TimeEnd { get; set; }
        public string Sort { get; set; }
        public bool IsPaging { get; set; } = true;
    }
}

File: eTierMeeting_v2/SPA/src/app/_core/_dtos/search-history-params.ts
export interface SearchHistoryParams {
    machineId: string;
    pdcId: number | null;
    buildingCode: number | null;
    cellName: number | null;
    positionCode: string;
    timeStart: string;
    timeEnd: string;
    sort: string;
    isPaging: boolean;
}
